frontend/src/components/header/Header.tsx

```diff
@@ -36,21 +47,16 @@ const Header: React.FC<HeaderProps> = ({isLogged}: HeaderProps) => {
     };
 
     const findRouteByPath = (path: string) => {
-        const normalizedPath = path.toLowerCase().replace(/\s/g, '');
+        const normalizedPath = path.toLowerCase().replace(/\s/g, '') === "home" ? "" :
+            path.toLowerCase().replace(/\s/g, '');
         return mainRoutes.find((route: IRoute) => route.path.trim().slice(1).toLowerCase() === normalizedPath);
     };
 
-    const redirectToPage = (isLogged: boolean, navigate: NavigateFunction) => {
-        if (isLogged) {
-            navigate(Paths.Home);
-        } else if (location.pathname !== Paths.SignUp) {
-            navigate(Paths.SignIn);
-        }
-    };
-
     useEffect(() => {
-        redirectToPage(isLogged, navigate);
-    }, [isLogged]);
+        setMainTabs(getInitialStateMainTabs(loggedUserType));
+        setSettingsTabs(getInitialStateSettingsTabs(loggedUserType));
+        setMainRoutes(getInitialState(loggedUserType));
+    }, [loggedUserType]);
```

Comment:
in what cases does user change his type?

Reply:
when i logged out and change userType, it doesn't refresh data, but if i manually reload it's updated